extensions/interactions/FractionInput/directives/FractionInputValidationServiceSpec.js

```diff
@@ -139,11 +172,84 @@ describe('FractionInputValidationService', function() {
     function() {
       answerGroups[0].rules = [greaterThanMinusOne, equalsOneRule];
       var warnings = validatorService.getAllWarnings(
-        currentState, {}, answerGroups, goodDefaultOutcome);
+        currentState, customizationArgs, answerGroups,
+        goodDefaultOutcome);
       expect(warnings).toEqual([{
         type: WARNING_TYPES.ERROR,
         message: 'Rule 2 from answer group 1 will never be matched ' +
           'because it is made redundant by rule 1 from answer group 1.'
       }]);
     });
+
+  it('should catch redundant rules caused by exactly equals', function() {
+    answerGroups[0].rules = [exactlyEqualToOneAndNotInSimplestFormRule];
+    var warnings = validatorService.getAllWarnings(
+      currentState, customizationArgs, answerGroups,
+      goodDefaultOutcome);
+    expect(warnings).toEqual([{
+      type: WARNING_TYPES.ERROR,
+      message: 'Rule 1 from answer group 1 will never be matched ' +
+        'because it is not in simplest form.'
+    }]);
+  });
+
+
+  // ========================== NEW RULES =================================
+  it('should catch non integer inputs in the numerator', function() {
+    answerGroups[0].rules = [nonIntegerRule];
+    var warnings = validatorService.getAllWarnings(
+      currentState, customizationArgs, answerGroups,
+      goodDefaultOutcome);
+    expect(warnings).toEqual([{
+      type: WARNING_TYPES.ERROR,
+      message: (
+        'Rule ' + 1 + ' from answer group ' +
+        1 + ' is invalid as input must be an ' +
+        'integer.')
+    }]);
+  });
+
+  it('should catch non integer inputs in the whole number', function() {
+    nonIntegerRule.type = 'HasIntegerPartEqualTo';
+    answerGroups[0].rules = [nonIntegerRule];
+    var warnings = validatorService.getAllWarnings(
+      currentState, customizationArgs, answerGroups,
+      goodDefaultOutcome);
+    expect(warnings).toEqual([{
+      type: WARNING_TYPES.ERROR,
+      message: (
+        'Rule ' + 1 + ' from answer group ' +
+        1 + ' is invalid as input must be an ' +
+        'integer.')
+    }]);
+  });
+
+  it('should catch non integer inputs in the denominator', function() {
+    nonIntegerRule.type = 'HasDenominatorEqualTo';
+    answerGroups[0].rules = [nonIntegerRule];
+    var warnings = validatorService.getAllWarnings(
+      currentState, customizationArgs, answerGroups,
+      goodDefaultOutcome);
+    expect(warnings).toEqual([{
+      type: WARNING_TYPES.ERROR,
+      message: (
+        'Rule ' + 1 + ' from answer group ' +
+        1 + ' is invalid as input must be an ' +
+        'integer.')
+    }]);
+  });
+
+  it('should catch non integer inputs in the numerator', function() {
```

Comment:
This description seems wrong.

Reply:
Should have been a divide by zero case, done.